authors/list_text.py

```diff
@@ -2,18 +2,66 @@
 
 from __future__ import print_function
 
+from io import open
 from json import load
 
 author_list = load(open("authors.json"))
+author_list = list(filter(lambda x: x["sympy_commits"] > 0, author_list))
 
-print("List of authors:")
-print()
-
-for author in author_list:
-    print("Author:", author["name"])
-    print("Affiliation:", author["institution"])
-    print("Address:", author["institution_address"])
-    print("GitHub profile: https://github.com/%s" % author["github_id"])
-    print("Article commits: https://github.com/sympy/sympy-paper/commits?author=%s" % author["github_id"])
-    print("Article comments on issues and PRs: https://github.com/sympy/sympy-paper/search?q=%s&type=Issues" % author["github_id"])
-    print()
+print("Generating authors.md")
+
+with open("authors.md", "w", encoding='utf-8') as f:
+    f.write(u"""\
+# Article Authors
+
+All authors satisfy the following ICMJE [criteria](http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html#two) for authorship:
+
+1. Substantial contributions to the conception or design of the work; or the
+   acquisition, analysis, or interpretation of data for the work; AND
+
+2. Drafting the work or revising it critically for important intellectual
+   content; AND
+
+3. Final approval of the version to be published; AND
+
+4. Agreement to be accountable for all aspects of the work in ensuring that
+   questions related to the accuracy or integrity of any part of the work are
+   appropriately investigated and resolved.
+
+
+The column `SymPy Commits` lists how many commits the authors contributed to
+SymPy (with a link to the actual commits), which shows how the requirement 1.
+is satisfied.
+
+The column `Article Contributions` lists all commits and comments to the
+article itself, which shows how the requirement 2. is satisfied.
+
+All authors approved the final version of the article, satisfying the
+requirement 3.
+
+Finally, all authors agreed to the requirement 4.
+
+""")
+    f.write(u"\n")
+    f.write(u"| Author | GitHub Profile | SymPy Commits | "
+            "Article Contributions | Affiliation |\n")
+    f.write(u"| :---   | :---           |     :---:     | "
+            ":---                  | :---        |\n")
+
+    for author in author_list:
+        f.write(u"| %s | [@%s](https://github.com/%s) | "
+            "[%d](https://github.com/sympy/sympy/commits?author=%s) | "
+            "[commits](https://github.com/sympy/sympy-paper/commits?author=%s),"
+                " [comments](https://github.com/sympy/sympy-paper/search"
+                "?q=%s&type=Issues) "
```

Comment:
For comments you should search for "commenter:asmeurer" (see https://help.github.com/articles/searching-issues/).